packages/flutter/lib/src/widgets/scroll_configuration.dart

```diff
@@ -2,12 +2,86 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-//  DELETE THIS FILE WHEN REMOVING LEGACY SCROLLING CODE
-
 import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
 
 import 'framework.dart';
 import 'scroll_behavior.dart';
+import 'scroll_physics.dart';
+import 'overscroll_indicator.dart';
+
+class ScrollBehavior2 {
+  const ScrollBehavior2();
+
+  /// The platform whose scroll physics should be implemented.
+  ///
+  /// Defaults to the current platform.
+  TargetPlatform getPlatform(BuildContext context) => defaultTargetPlatform;
+
+  /// The color to use for the glow effect when [platform] indicates a platform
+  /// that uses a [GlowingOverscrollIndicator].
+  ///
+  /// Defaults to white.
+  Color getGlowColor(BuildContext context) => const Color(0xFFFFFFFF);
```

Comment:
we should move this to an inherited widget that is wrapped around the glow by the material layer (the glow itself coming from this layer)